src/lib/features/authenticate/model/sign-in-snackbar.ts

```diff
@@ -14,5 +17,7 @@ export function signInSnackbar() {
     });
     url.searchParams.delete('snackbar');
     window.history.replaceState(null, '', url.toString());
+    get(likesStore).forEach(addCloudLike);
```

Comment:
question: why did you decide to do it here, of all the places?)

suggestion: the same as in that other PR, let's do it here:

https://github.com/aabounegm/cast/blob/bd8fc73fbd98ddfbea73112d6fdbc42413c7d46a/src/lib/features/like-episode/model/like.ts#L13-L18

Reply:
Because I didn't want to add the like to Supabase every time the user loads the page (I'm not sure if there are any uniqueness constraints set in the backend). This seemed like the only function where I guarantee the user has just signed in after being logged out before and having added some episode to the favorites.
I agree it's not the best solution, especially that it only works if the user signed in through the snackbar that appears after liking an episode unauthenticated, but I also couldn't think of another quick solution that doesn't involve querying the backend likes/history and diffing the data.

Reply:
There are uniqueness constraints on the data, so uploading a duplicate like/history won't really change anything except for the `created_at` timestamp, which we don't use anyway, except for ordering.

Technically, we could have another `derived` store that tracks the previous value of the `user` store, and expose to the public yet another `derived` store that provides access to them both, however, that is slightly janky, so proceed at will

Reply:
Still, I would prefer not to depend on constraints in the backend, especially that the function name literally says "add". It's better to validate the data on the client side first and consider the backend validation as a safety measure

Reply:
Alright, then diffing on the client it is. Shouldn't be too expensive/hard